service.go

```diff
@@ -88,6 +88,23 @@ const (
 	optionLaunchdConfig = "LaunchdConfig"
 )
 
+// Status represents service status as an interger value
+type Status int
+
+// Status of service represented as an integer
+const (
+	StatusNotImplemented Status = iota
+	StatusUnknown
+	StatusError
```

Comment:
What if the StatusError was combined into StatusUknown? The fewer possible states the consumer has to handle is probably better.